### PR TITLE
Update testing utils

### PR DESCRIPTION
**Pull request by adisky**
**What type of PR is this?**

 /kind code-refactoring

**What does does this PR do / why we need it**:
Fixes # fixing unit tests of adpaters/common after update to devfile V2

**Which issue(s) this PR fixes**:

**How to test changes / Special notes to the reviewer**:

**NOTE: PR for parser-v2 branch**

### CODE REVIEW

File: pkg/testingutil/devfile.go [hunk 1]
```diff
@@ -7,7 +7,7 @@ import (
 // TestDevfileData is a convenience data type used to mock up a devfile configuration
 type TestDevfileData struct {
 	ComponentType       versionsCommon.DevfileComponentType
-	CommandActions      []versionsCommon.DevfileCommandAction
+	CommandActions      []versionsCommon.Exec
```
Comment (mik-dass) [inline pkg/testingutil/devfile.go:10]: let's rename this

File: pkg/devfile/parser/writer_test.go [hunk 2]
```diff
@@ -23,9 +22,9 @@ func TestWriteJsonDevfile(t *testing.T) {
 		devfileObj := DevfileObj{
 			Ctx: devfileCtx.NewDevfileCtx(devfileTempPath),
 			Data: &v100.Devfile100{
-				ApiVersion: common.ApiVersion(apiVersion),
-				Metadata: common.DevfileMetadata{
-					Name: &testName,
+				ApiVersion: v100.ApiVersion(apiVersion),
```
Comment (mik-dass) [inline pkg/devfile/parser/writer_test.go:25]: do we need a test for v2

File: pkg/testingutil/devfile.go (hunk 1 above)
Comment (adisky) [reply pkg/testingutil/devfile.go]: Done

File: pkg/devfile/parser/writer_test.go (hunk 2 above)
Comment (adisky) [reply pkg/devfile/parser/writer_test.go:25]: Will add in a separate PR

**Comment by amitkrout** [comment]
Is the pr open for review ?. @adisky Can you please add description/link to this pr like
```why we need this ?```.

**Comment by adisky** [comment]
@amitkrout Sorry I forgot to add the note, this pr is for `parser-v2` branch.

File: pkg/devfile/adapters/common/command_test.go [hunk 3]
```diff
@@ -10,215 +10,126 @@ import (
...
-			commandActions: []common.DevfileCommandAction{
-				{
-					Command:   &commands[0],
-					Component: &invalidComponent,
-					Type:      &validCommandType,
-				},
-			},
-			isCommandRequired: []bool{true},
-			wantErr:           true,
+			requestedType: []common.DevfileCommandGroupType{runGroup},
```
Comment (mik-dass) [inline pkg/devfile/adapters/common/command_test.go:107]: a case for group mismatch too

File: pkg/devfile/adapters/common/command_test.go [hunk 4]
```diff
@@ -10,215 +10,126 @@ import (
...
-					ComponentType:  common.DevfileComponentTypeDockerimage,
+					ExecCommands: tt.execCommands,
+					Components:   components,
 				},
 			}
 
-			for i, commandName := range tt.requestedCommands {
-				command, err := getCommand(devObj.Data, commandName, tt.isCommandRequired[i])
+			for _, gtype := range tt.requestedType {
+				_, err := getCommand(devObj.Data, "", gtype)
```
Comment (mik-dass) [inline pkg/devfile/adapters/common/command_test.go:125]: the command name shouldn't be empty

File: pkg/devfile/adapters/common/command_test.go [hunk 5]
```diff
@@ -10,215 +10,126 @@ import (
...
-			name:              "Case 11: Valid devfile with empty workdir",
-			requestedCommands: []string{"devrun"},
-			commandActions: []common.DevfileCommandAction{
+			name: "Case 7: Invalid command referencing an absent component",
+			execCommands: []common.Exec{
 				{
-					Command:   &commands[0],
-					Component: &components[0],
-					Type:      &validCommandType,
+					CommandLine: commands[0],
```
Comment (mik-dass) [inline pkg/devfile/adapters/common/command_test.go:102]: a test case where multiple commands of a type exist but the default is returned

File: pkg/devfile/adapters/common/command_test.go (hunk 4 above)
Comment (adisky) [reply pkg/devfile/adapters/common/command_test.go:125]: this scenario is covered for all three commands, check tests for `TestGetBuildCommand` `https://github.com/openshift/odo/pull/3220/files#diff-0e2dae7fac15ec15148977126c96a904R346`

Comment (mik-dass) [reply pkg/devfile/adapters/common/command_test.go]: OK

File: pkg/devfile/adapters/common/command_test.go (hunk 5 above)
Comment (adisky) [reply pkg/devfile/adapters/common/command_test.go:104]: Added

File: pkg/devfile/adapters/common/command_test.go (hunk 3 above)
Comment (adisky) [reply pkg/devfile/adapters/common/command_test.go:109]: Done

**Pull request merged**